pkg/engine/task/pod_exec.go

```diff
@@ -0,0 +1,87 @@
+package task
+
+import (
+	"fmt"
+	"io"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+// PodExec defines a command that will be executed in a running Pod.
+// RestCfg - The REST configuration for the cluster.
+// PodName - The pod name on which to execute the command.
+// PodNamespace - Namespace of the pod
+// ContainerName - optional container to execute the command in. If empty, first container is taken
+// Args - The command (and args) to execute.
+// In - An (optional) command input stream.
+// Out - The command output stream set by `Run()`.
+// Err - the command error stream set by `Run()`.
+type PodExec struct {
+	RestCfg       *rest.Config
+	PodName       string
+	PodNamespace  string
+	ContainerName string
+	Args          []string
+	In            io.Reader
+	Out           io.Writer
+	Err           io.Writer
+}
+
+// Run execution the pod command.
+func (pe *PodExec) Run() error {
+	codec := serializer.NewCodecFactory(scheme.Scheme)
+	restClient, err := apiutil.RESTClientForGVK(
+		schema.GroupVersionKind{
+			Version: "v1",
+			Kind:    "pods",
+		},
+		pe.RestCfg,
+		codec)
+	if err != nil {
+		return err
+	}
+
+	req := restClient.
+		Post().
+		Resource("pods").
+		Name(pe.PodName).
+		Namespace(pe.PodNamespace).
+		SubResource("exec")
+
+	poe := &v1.PodExecOptions{
```

Comment:
peo? :D if we need to be weird like that 😝

Reply:
oh, this is very well within [Go naming guidelines](https://github.com/golang/go/wiki/CodeReviewComments#variable-names) 
```
Variable names in Go should be short rather than long. This is especially true for local variables with limited scope. Prefer c to lineCount. Prefer i to sliceIndex.

The basic rule: the further from its declaration that a name is used, the more descriptive the name must be. For a method receiver, one or two letters is sufficient. Common variables such as loop indices and readers can be a single letter (i, r). More unusual things and global variables need more descriptive names.
```
here, `poe` is local and used only once, directly after the declaration 👅

Reply:
I think @alenkacz was more about `peo` vs `poe`, because of **P**od**E**xec**O**ptions ;)

Reply:
yeah, I am not a big fan of these names (peo or poe is very awkward to me) but we should at least fix that random permutation of letters :) thanks @ANeumann82

Reply:
Andreas is right! It should've been `peo` but I got rid of it altogether so 🤷‍♂